compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BStructureType.java

```diff
@@ -49,4 +54,22 @@ public BStructureType(int tag, BTypeSymbol tSymbol, int flags) {
     public void accept(TypeVisitor visitor) {
         visitor.visit(this);
     }
+
+    protected boolean shouldPrintShape(Name name) {
+        if (name == null) {
+            return false;
+        }
+
+        String value = name.value;
+
+        if (value.isEmpty() || value.startsWith(DOLLAR)) {
+            return true;
+        }
+
+        if (!(Symbols.isFlagOn(this.tsymbol.flags, Flags.READONLY))) {
+            return false;
+        }
+
+        return value.startsWith("($");
```

Comment:
We have already checked part of this condition above L65. Should we return `false` here?